File: app/src/main/java/com/companyname/kotlinpractice/LoginViewModel.java
package com.companyname.kotlinpractice;

import androidx.lifecycle.LiveData;
import androidx.lifecycle.MediatorLiveData;
import androidx.lifecycle.MutableLiveData;
import androidx.lifecycle.ViewModel;

import java.util.ArrayList;

import io.reactivex.rxjava3.core.Observable;

public class LoginViewModel extends ViewModel {

    public MutableLiveData<ArrayList<Coin>> ldCoin = new MutableLiveData<>();

    public MutableLiveData<String> name = new MutableLiveData<>("test");
    public MutableLiveData<String> password = new MutableLiveData<>("123456");

//    MediatorLiveData ===== CombineLatest
    public MediatorLiveData<Boolean> btnEnable = new MediatorLiveData<>();

    LoginViewModel() {
        btnEnable.addSource(name, s -> {
            checkBtnEnable();
        });
        btnEnable.addSource(password, s -> {
            checkBtnEnable();
        });
    }

    private void checkBtnEnable() {
        if (name.getValue() == null || password.getValue() == null) {
            btnEnable.setValue(false);
            return;
        } else if (name.getValue().length() == 0 || password.getValue().length() == 0) {
            btnEnable.setValue(false);
            return;
        }
        else {
            btnEnable.setValue(true);
            return;
        }

    }

    public Observable<ArrayList<Coin>> getCoins() {
        return CoinRepo.getCoins("USD")
//                .onErrorReturnItem()
                .doOnNext(coins -> ldCoin.postValue(coins));
    }
}

File: app/src/main/java/com/companyname/kotlinpractice/LoginViewModel.kt
package com.companyname.kotlinpractice

import androidx.lifecycle.LiveData
import androidx.lifecycle.MediatorLiveData
import androidx.lifecycle.MutableLiveData
import androidx.lifecycle.ViewModel

import java.util.ArrayList

import io.reactivex.rxjava3.core.Observable

class LoginViewModel() : ViewModel() {

    var ldCoin = MutableLiveData<ArrayList<Coin>>()

    var name = MutableLiveData<String>("test")
    var password = MutableLiveData<String>("123456")

//    MediatorLiveData ===== CombineLatest
    var btnEnable = MediatorLiveData<Boolean>()

    init {
        btnEnable.addSource(name) { checkBtnEnable() }
        btnEnable.addSource(password) { checkBtnEnable() }
    }

    private fun checkBtnEnable() {
        btnEnable.value = !(name.value.isNullOrBlank() || password.value.isNullOrBlank())
    }

    fun getCoins() : Observable<ArrayList<Coin>> {
        return CoinRepo.getCoins("USD")
//                .onErrorReturnItem()
                .doOnNext { ldCoin.postValue(it) }
    }
}
